### ISSUE TITLE
Fix vector of bool (add correct overload)

### ISSUE THREAD

**Issue by lifflander** [DARMA-tasking/checkpoint#71]
`std::vector<bool>` has special semantics. A general serializer, as written in checkpoint, can't be used to serialize that vector because you can't take a reference to a single element (i.e., bit iterator `__bit_iterator`). Read this article: [Vector of Bool](https://isocpp.org/blog/2012/11/on-vectorbool)

We need to write an overload for vector of bool in checkpoint so we can correctly serialize them. Add a test that exposes this problem in the checkpoint testing so it doesn't come up again.

**Comment by mperrinel** [comment]
@lifflander 

I implemented in the _vector_serialize.h_ of checkpoint Lib : 
```
template <typename Serializer, typename VectorAllocator>
void serialize(Serializer& s, std::vector<bool, VectorAllocator>& vec) {
  serializeVectorMeta(s, vec);
  for (bool elt : vec) {
    s | elt;
  }
}
```
The idea is to specialise the method : 
```
template <typename Serializer, typename T, typename VectorAllocator>
void serialize(Serializer& s, std::vector<T, VectorAllocator>& vec) {
  serializeVectorMeta(s, vec);
  serializeArray(s, &vec[0], vec.size());
}
```
which cannot work for a vector of bool.

If my code compile, it doesn't work for the new test I have created. This test deserialize only some false values.
The problem comes from this line : 
`    s | elt;
`
elt contains correctly false or true depending of my test but it's not correctly serialized.
Of course I cannot give elt as a reference because I will go back into the same problem of bool &.
I tried to do the same with a vector of int and I saw that if, during the loop, I serialize a copy of the value instead of the reference, that doesn't work anymore. Does that mean the serializer keeps the reference on the local variable which is destroyed instead of doing a copy ?
